test/testuninitvar.cpp

```diff
@@ -4389,6 +4390,17 @@ class TestUninitVar : public TestFixture {
         ASSERT_EQUALS("[test.cpp:8] -> [test.cpp:4]: (warning) Uninitialized variable: i\n", errout_str());
     }
 
+    void valueFlowUninit_lambda() {  // #12944
+        valueFlowUninit( "void Fun()\n"
+                         "{\n"
+                         "    int var;\n"
+                         "    auto lam = [&]() { int x = var; };\n"
+                         "    var = 5;\n"
```

Comment:
I assume after your changes that no error will be written if `var = 5;` is removed?

It feels like a better fix would be in valueflow? I assume it "executes" the lambda when it is defined. It should probably skip the lambda first and then when `lam()` is called it can "execute" the lambda.